### PR TITLE
 Synchronize IME insets with iOS keyboard

### PR DESCRIPTION
**Pull request by terrakok**
Proper support of the keyboard animation

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/KeyboardOverlapHeight.uikit.kt [hunk 1]
```diff
@@ -22,6 +22,6 @@ import androidx.compose.runtime.*
  * Composition local for height that is overlapped with keyboard over Compose view.
  */
 @InternalComposeApi
-val LocalKeyboardOverlapHeightState = staticCompositionLocalOf<State<Float>> {
+val LocalKeyboardOverlapHeightState = staticCompositionLocalOf<Float> {
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/KeyboardOverlapHeight.uikit.kt]: Conflict with #873
~Let's merge that one first.~ Done

**Comment by dima-avdeev-jb** [comment]
@terrakok Can you please merge latest changes from jb-main branch

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt [hunk 2]
```diff
@@ -156,6 +161,15 @@ internal actual class ComposeWindow : UIViewController {
     private var safeArea by mutableStateOf(PlatformInsets())
     private var layoutMargins by mutableStateOf(PlatformInsets())
 
+    //invisible view to track system keyboard animation
+    private val keyboardAnimationView: UIView by lazy {
+        UIView.new()!!.apply {
+            setFrame(CGRectMake(0.0, 0.0, 0.0, 0.0))
+            hidden = true
+        }
```
Comment (dima-avdeev-jb) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt]:
```suggestion
        UIView(CGRectMake(0.0, 0.0, 0.0, 0.0)).apply {
            hidden = true
        }
```
I don't know which variant is better.
Why are you using UIView.new()!! ?

**Review by dima-avdeev-jb — APPROVED** [review]

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt (hunk 2 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt]: My vote is to avoid `!!`

Comment (elijah-semyonov) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt]: I'd avoid this `new` method. Cinterop can't derive correct nullability for it.

**Comment by elijah-semyonov** [comment]
LGTM, the only question is `UIView.animateWithDuration` default curve is correct?
See https://developer.apple.com/documentation/uikit/uiresponder/1621613-keyboardanimationcurveuserinfoke

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt (hunk 2 above)
Comment (terrakok) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt]: I didn't notice the explicit constructor :)

**Comment by terrakok** [comment]
Regarding the default curve, as I understood, default is linear and it is good for us. We can extract it from the keyboard options but it is not required, I guess

**Pull request merged**